### ISSUE TITLE
 Possible edge case in decoder when very first OP is QOI_OP_RUN

### ISSUE THREAD

**Issue by phoboslab** [aldanor/qoi-rust#5]
There's an edge-case that currently is handled differently by various decoders. At a first glance it looks like this decoder may not be spec-conformant. The spec is not clearly worded in this case, but I have decided to interpret it in a particular way now. Sorry about that!

Please see the original issue here https://github.com/phoboslab/qoi/issues/258 for the details.

**Comment by AregevDev** [comment]
Hi, this shouldn't be hard to fix. Just add the starting pixel color to the index array.

I have a a question for @phoboslab though:
Does the spec allow the encoder to start with a run operation? You mentioned you never do that in the reference encoder, but other encoder implementations may do.

@aldanor If that's an acceptable solution, I can open a PR in the coming days. Also, I have a few specific questions about the implementation of the encoder / decoder. How can I reach out?

**Comment by aldanor** [comment]
Yes; should add the `edgecase.qoi` to the current test suite and see what happens. Should be an easy fix if it's misinterpreted.

**Comment by AregevDev** [comment]
I wrote a test that reads `edgecase.qoi` and decodes it and works as intended.

I still need to debug the decoding process though, I am not sure when it puts the initial opaque black pixel in the `seen` array.

**Comment by aldanor** [comment]
Test added in #12, it seems that the decoder was already acting in compliant way (first pixel being black opaque).

**Comment by pearcebasmanm** [comment]
I created a new issue relating to this before noticing this closed one, but I'm not sure if this is actually resolved. The test added does not cover this problem, since it only arises when an INDEX operation happens later that references the color. Eg: `RUN 1, RBA 255 255 255, INDEX 53` should result in `Black, White, Black`, but ends up being `Black, White, Transparent` since the opaque black is never put in the index.

**Comment by aldanor** [comment]
Another edge case addressed in #17 (by @Martomate), now merged in.
